packages/ovault-evm/contracts/OVaultComposer.sol

```diff
@@ -154,9 +154,9 @@ contract OVaultComposer is IOVaultComposer, ReentrancyGuard {
     /// @dev Permissionless function to send back the message to the source chain
     /// @dev Always possible unless the lzCompose() fails due to an Out-Of-Gas panic
     function refund(bytes32 _guid, bytes calldata _extraOptions) external payable nonReentrant {
-        if (failedGuidState(_guid) != FailedState.CanOnlyRefund) revert CanNotRefund(_guid);
-
         FailedMessage memory failedMessage = failedMessages[_guid];
+        if (failedMessage.refundOFT != address(0)) revert CanNotRefund(_guid);
+
```

Comment:
see https://github.com/LayerZero-Labs/devtools/pull/1600/commits/f4f5703eb7cd2b7280d9f69ce2c76ad7b8b3a073#r2184625897

also if refundOFT!=0 then failedMessage.oft could still be 0

so probably should be:
```diff
-address refundOft = failedMessage.oft;
+address refundOft = failedMessage.refundOFT;
```